staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -121,6 +132,197 @@ const (
+
+	// Validations contain CEL expressions which is used to apply the validation.
+	// Required.
+	Validations []Validation `json:"Validations,omitempty" protobuf:"bytes,3,rep,name=Validations"`
+}
+
+// ParamSource is a tuple of Group Kind and Version.
+type ParamSource struct {
+	// APIGroup is the API group the resources belong to. '*' is all groups.
+	// Required.
+	APIGroup string `json:"apiGroup,omitempty" protobuf:"bytes,1,rep,name=apiGroup"`
+
+	// APIVersion is the API version the resources belong to. '*' is all versions.
+	// Required.
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,2,rep,name=apiVersion"`
+
+	// APIKind is the API kind the resources belong to. '*' is all versions.
+	// Required.
+	APIKind string `json:"apiKind,omitempty" protobuf:"bytes,3,rep,name=apiKind"`
+}
+
+// Validation specified the CEL expression which is used to apply the validation.
+type Validation struct {
+	// The name of the validation rule.
+	// Required.
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// The Expression for validation.
+	// Required.
+	Expression string `json:"expression" protobuf:"bytes,2,opt,name=Expression"`
+	// The Message retured when failed the validation.
+	// Optional.
+	Message string `json:"message" protobuf:"bytes,3,opt,name=message"`
+	// Enforcement defines how errors from validation checks are handled -
+	// allowed values are Deny or Warn. Defaults to Deny.
+	// +optional
+	Enforcement *EnforcementType `json:"enforcement,omitempty" protobuf:"bytes,4,opt,name=enforcement,casttype=EnforcementType"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PolicyBinding describes the configuration of an admission validation policy.
+type PolicyBinding struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Specification of the desired behavior of the AdmissionPolicy.
+	// +optional
+	Spec PolicyBindingSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PolicyBindingList is a list of PolicyBinding.
+type PolicyBindingList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// List of PolicyBinding.
+	Items []PolicyBinding `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PolicyBindingSpec is the specification of the PolicyBinding.
+type PolicyBindingSpec struct {
+	//Policy specifies the admission control policy definition which the Param used on
+	// Required.
```

Comment:
`references a ValidatingAdmissionPolicy`?

Note that this ref is for more than assigning params.